test/js/search/search-query-display.test.js

```diff
@@ -1,61 +1,79 @@
-import React from 'react';
-import * as ReactAll from 'react';
-import { mount } from 'enzyme';
+import React from 'react'
+import * as ReactAll from 'react'
+import { mount } from 'enzyme'
 
-const fetch = require('node-fetch');
+const fetch = require('node-fetch')
+
+import SearchQueryDisplay, { ClearAllButton } from 'components/SearchQueryDisplay'
+import { PropsStudySearchProvider } from 'components/search/StudySearchProvider'
+import KeywordSearch from 'components/KeywordSearch'
 
-import SearchQueryDisplay from 'components/SearchQueryDisplay';
 
 const oneStringFacet = [
-  {id: 'species', filters: [{id: "NCBITaxon_9606", name: "Homo sapiens"}]}
+  { id: 'species', filters: [{ id: 'NCBITaxon_9606', name: 'Homo sapiens' }] }
 ]
 
 const twoStringFacets = [
-  {id: 'disease', filters: [{id: "id1", name: "disease1"}]},
-  {id: 'species', filters: [{id: "NCBITaxon_9606", name: "Homo sapiens"}]}
+  { id: 'disease', filters: [{ id: 'id1', name: 'disease1' }] },
+  { id: 'species', filters: [{ id: 'NCBITaxon_9606', name: 'Homo sapiens' }] }
 ]
 
 const stringAndNumericFacets = [
-  {id: 'species', filters: [
-    {id: "NCBITaxon_9606", name: "Homo sapiens"},
-    {id: "NCBITaxon_10090", name: "Mus musculus"}
-  ]},
-  {id: "organism_age", filters: {min: 14, max: 180, unit: "years"}}
+  {
+    id: 'species', filters: [
+      { id: 'NCBITaxon_9606', name: 'Homo sapiens' },
+      { id: 'NCBITaxon_10090', name: 'Mus musculus' }
+    ]
+  },
+  { id: 'organism_age', filters: { min: 14, max: 180, unit: 'years' } }
 ]
 
 describe('Search query display text', () => {
   it('renders a single facet', async () => {
     const wrapper = mount((
       <SearchQueryDisplay facets={oneStringFacet} terms={''}/>
     ))
-    expect(wrapper.text().trim()).toEqual(': Metadata contains (species: Homo sapiens)')
+    expect(wrapper.text().trim()).toEqual(': Metadata contains (species: Homo sapiens) Clear All')
   })
 
   it('renders multiple facets', async () => {
     const wrapper = mount((
       <SearchQueryDisplay facets={twoStringFacets} terms={''}/>
     ))
-    expect(wrapper.text().trim()).toEqual(': Metadata contains (disease: disease1) AND (species: Homo sapiens)')
+    expect(wrapper.text().trim()).toEqual(': Metadata contains (disease: disease1) AND (species: Homo sapiens) Clear All')
```

Comment:
this makes a good case for either adding a prop 'showClearAll'  to SearchQueryDisplay that can be used to hide the button so that you can isolate the components for testing, or rendering the button from ResultsPanel.js instead

Reply:
I'm not sure how you would test this seeing that the button always appears when the SearchQuery is mounted.

Reply:
Exactly, if you have showClearAll as a prop, then you can set it to false in the test, and not have to worry about exactly how the button renders when we're testing other things.  Alternatively, you can wrap the text part of the display in a separate span, so you can just get that text